### PR TITLE
Add ubuntu 15.10 box

### PR DESCRIPTION
**Pull request by ppp0**
In "plain" and "default" flavours.

### CODE REVIEW

**Comment by ppp0** [comment]
As I said, ymmw still

File: ubuntu-1510/scripts/puppet.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Prepare puppetlabs repo
+wget http://apt.puppetlabs.com/puppetlabs-release-wheezy.deb
```
Comment (njam) [inline ubuntu-1510/scripts/puppet.sh]: Is this the correct deb?

**Comment by ppp0** [comment]
@njam @kris-lab please take note

As discussed with Chris, I will suspend work on wily 15.10 - the kernel there is v4.2.0-16 and this is going to make troubles when building the Blackmagic/Decklink driver

Doing a vivid 15.04 version

**Comment by ppp0** [comment]
Spec tests passing for both ubuntu-15.10 plain and default flavs

**Comment by ppp0** [comment]
@njam please review

Follow-up: https://github.com/cargomedia/vagrant-boxes/issues/54

**Comment by njam** [comment]
- plain.json: Missing aws builder
- grub.sh: What about this `exit 0` - not needed? remove this script?
- base.sh: Why not install unzip?
- virtualbox-vagrant.sh: Why not install nfs client?
- Could you update the readme with the new available boxes?

**Comment by ppp0** [comment]
* Re-added aws builder, sorry
* exit 0 removed :)
* see next
* unzip and nfs-client are now being installed via preseed.cfg
* ok

**Comment by njam** [comment]
- Need to adjust the source AMIs, now it's Debian wheezy ones which will be used to build

**Comment by njam** [comment]
https://cloud-images.ubuntu.com/releases/wily/beta-2/
http://cloud-images.ubuntu.com/releases/15.04/release/

Build and check AMI build.

**Comment by ppp0** [comment]
Did not succeed so far. I could build a box using this builder params:
```json
    {
      "name": "aws",
      "type": "amazon-ebs",
      "region": "eu-west-1",
      "source_ami": "ami-13436a64",
      "instance_type": "m3.large",
      "ssh_username": "ubuntu",
      "ami_name": "ubuntu-15.10-amd64-default {{timestamp}}",
      "ami_description": "https://github.com/cargomedia/vagrant-boxes",
      "ami_groups": ["all"],
      "ami_regions": ["us-east-1"],
      "ssh_timeout": "5m"
    }
```

I confirm I can launch and log in into `ami-13436a64` (eu-west-1, hvm - see https://cloud-images.ubuntu.com/releases/wily/beta-2/)

After running the build, I can't log in remotely to the instance running the generated AMI

* tried `eu-west-1` and `us-east-1`
* tried different types of instance, mostly using `m3.medium` though
* ping not working

Inspecting the system log:
```
   29.330678] cloud-init[600]: Cloud-init v. 0.7.7 running 'init' at Wed, 21 Oct 2015 15:03:13 +0000. Up 28.75 seconds.

[   29.335192] cloud-init[600]: ci-info: +++++++++++++++++++++++++++Net device info++++++++++++++++++++++++++++

[   29.335334] cloud-init[600]: ci-info: +--------+-------+-----------+-----------+-------+-------------------+

[   29.335470] cloud-init[600]: ci-info: | Device |   Up  |  Address  |    Mask   | Scope |     Hw-Address    |

[   29.335603] cloud-init[600]: ci-info: +--------+-------+-----------+-----------+-------+-------------------+

[   29.335736] cloud-init[600]: ci-info: |  eth0  | False |     .     |     .     |   .   | 22:00:0a:4d:17:da |

[   29.335878] cloud-init[600]: ci-info: |   lo   |  True | 127.0.0.1 | 255.0.0.0 |   .   |         .         |

[   29.336011] cloud-init[600]: ci-info: |   lo   |  True |  ::1/128  |     .     |  host |         .         |

[   29.336176] cloud-init[600]: ci-info: +--------+-------+-----------+-----------+-------+-------------------+

[   29.336310] cloud-init[600]: 2015-10-21 15:03:14,449 - url_helper.py[WARNING]: Calling 'http://169.254.169.254/2009-04-04/meta-data/instance-id' failed [0/120s]: request error [('Connection aborted.', OSError(101, 'Network is unreachable'))]

[   30.340019] cloud-init[600]: 2015-10-21 15:03:15,458 - url_helper.py[WARNING]: Calling 'http://169.254.169.254/2009-04-04/meta-data/instance-id' failed [1/120s]: request error [('Connection aborted.', OSError(101, 'Network is unreachable'))]

```

Looks to me like somehow `eth0` was not assigned a correct ip address. DHCP failure?

**Comment by njam** [comment]
Maybe we break something when provisioning the box with our shell scripts?
How about trying the same without any provisioners?

Here's another ubuntu example, with a very basic provisioner:
Packer template: https://github.com/octohost/ubuntu-14.04/blob/master/template.json
Provisioner script: https://github.com/octohost/ubuntu-14.04/blob/master/scripts/setup.sh

**Comment by ppp0** [comment]
As discussed, let's postpone this until 15.10 is really needed (and officially released). Right now, its kernel version (4.x) breaks some of our tools

**Comment by njam** [comment]
@ppp0 closeable for now?

**Comment by njam** [comment]
@ppp0 what does it mean? :D

**Comment by ppp0** [comment]
Agree, closeable!

**Pull request closed without merging**